### ISSUE TITLE
To change chamber fan percentage i need to open and the panel many times

### ISSUE THREAD

**Issue by pitops** [bambulab/BambuStudio#1190]
**Describe the bug**
For some reason my printer keeps running at 100% chamber fan for ALL prints EVERY time. (printing PLA btw)

I go to manually reduce it after it starts getting to 100%. However with the latest bambu studio version **1.4.2.13** this became even more cumbersome.

First of all it takes time for the percentage to show in the settings as seen below all are zero.

![Screenshot 2023-01-27 at 2 19 15 PM](https://user-images.githubusercontent.com/9624361/215085459-9ec47934-2c62-4c1c-a56a-0e75a2838a2c.png)

![Screenshot 2023-01-27 at 2 19 22 PM](https://user-images.githubusercontent.com/9624361/215085481-08ea2791-7310-4579-9472-7b8a80cd8dc4.png)

After the stats come back. When i click on **-(minus)** it closes the panel and have to reopen to further reduce the speed from 100% to 50%. So i need to open the panel 5 times. 

![Screenshot 2023-01-27 at 2 21 11 PM](https://user-images.githubusercontent.com/9624361/215085803-83702803-3d9d-4c0c-bfcb-9138c98e8b63.png)

**To Reproduce**
1. Go to **Device** tab then see screenshots above

**Expected behavior**
1. Be able to change the chamber speed much faster (even be able to add my own value instead of requiring 5 clicks to go from 100% to 50%).
2. Avoid closing the panel when i hit **-** or **+**
3. If the chamber fan goes to 100% because of bambu studio and not because of printer firmware also fix that.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version 12.6

**Comment by walterwongbbl** [comment]
Hi.
This is a bug,     +or - operation should not close the window. 
 we will fix this issue.

**Comment by rkarlsba** [comment]
I see the same here with BS 1.4.2.13 and X1 Carbon with 01.04.00.00 firmware. If I click a fan, the three fan controls show up and if I choose to change one of them, the dialog closes and that fan is changed. I cannot find any way to set the chamber fan in the job either, so this must be done in realtime, making it pretty far from convenient. Setting the wanted chamber temperature also doesn't work or is simply ignored. We need to be able to control these things, really. The defaults make nice headlines, but they don't make a winner!

**Comment by walterwongbbl** [comment]
> I see the same here with BS 1.4.2.13 and X1 Carbon with 01.04.00.00 firmware. If I click a fan, the three fan controls show up and if I choose to change one of them, the dialog closes and that fan is changed. I cannot find any way to set the chamber fan in the job either, so this must be done in realtime, making it pretty far from convenient. Setting the wanted chamber temperature also doesn't work or is simply ignored. We need to be able to control these things, really. The defaults make nice headlines, but they don't make a winner!

Hi.
We will fix this issue. Can you provide a video of the operation. After clicking the button(+ or -), the fan control window will be closed.

**Comment by rkarlsba** [comment]
There's no need to make a video. Just open BS and click Device, then click one of the fan icons on the right, below the speed switch and the lights switch. Then attempt to change the speed on one of the fans and it'll close the dialogue, usually, but not always, changing the speed after closing it. It is NO way to set a predefined chamber temperature, probably because there isn't a chamber heater, only the bed and hotend heaters. Also, there are fans that can be used to control the amount of air going through the printer at any time, thus making it quite easy to control the chamber temperature. 

If this printer was open source, these bugs would have been fixed ages ago, but since you have chosen to stay closed, we customers must all beg you to fix them instead of just fixing the bugs ourselves and offering a pull request. The mentioned method works very well in thousands of open source systems and would work just as well here, leaving your precious workers' time to be used for other things. 

So please, can you open up now once and for all so that we don't get all this time waiting for someone to fix trivial bugs? 

Vennlig hilsen 

roy 
-- 
Roy Sigurd Karlsbakk 
(+47) 98013356 
-- 
I all pedagogikk er det essensielt at pensum presenteres intelligibelt. Det er et elementært imperativ for alle pedagoger å unngå eksessiv anvendelse av idiomer med xenotyp etymologi. I de fleste tilfeller eksisterer adekvate og relevante synonymer på norsk.

**Comment by DapperDangus** [comment]
You bought a closed source knowing so maybe don’t complain to the company working their *** off to deliver everything you guys want. Dear lord people, don’t buy a Mac then expect to modify it like a PC.

**Comment by rkarlsba** [comment]
I just suggest we could help out fixing the bugs they aren't able to fix if they were to open their systems. 

PS: A PC is a common word for any sort of computer. A Mac is a peculiar thing that can run mostly all sorts of OSes, but also macOS. 

Vennlig hilsen 

roy 
-- 
Roy Sigurd Karlsbakk 
(+47) 98013356 
-- 
I all pedagogikk er det essensielt at pensum presenteres intelligibelt. Det er et elementært imperativ for alle pedagoger å unngå eksessiv anvendelse av idiomer med xenotyp etymologi. I de fleste tilfeller eksisterer adekvate og relevante synonymer på norsk.

**Comment by DapperDangus** [comment]
In America it’s PC vs Mac but thanks…

Keep up the good work Bambu.

**Comment by walterwongbbl** [comment]
Hi.
Thank you for your suggestion. We have fixed this issue and will add it to the next version

**Comment by cameo69** [comment]
Please fix this. Also the fact that you have to wait until the current value has been loaded again.
![output_fan01](https://user-images.githubusercontent.com/44269764/218393445-00fb9d77-3c13-435b-b44f-aa5f168d8042.gif)
BambuStudio v1.4.2.13 on MacOS 13.1

**Comment by SaltWei** [comment]
Available in v1.5.0 Public Beta

**Comment by magnusmyklebost** [comment]
> Available in v1.5.0 Public Beta


When will this be added to the non beta? Im only using PLA and reducing chamber fan speed from 100% to 70% reduses the noise drastically. Ive never had any print issues.

I would like to be able to add a chamber fan value to the filament properties. Best would be to add it here, where you could set chamber fan, and also a field to "Automatic adjust fan speed if temperature goes above XXX degree" :
![image](https://user-images.githubusercontent.com/36155808/229752052-3bc1c5d2-fbeb-4ae1-a8ae-bd6ef3753c80.png)

**Comment by Felix-Quehl** [comment]
# Control Fan Speed in gcode of nozzle settings

![image](https://user-images.githubusercontent.com/67843709/236644817-502172fb-2930-4b82-81d9-31760bb1295e.png)

## M106 command

You can overwrite the camber fan speed with the gcode of the nozzle settings inside Bambu Studio:

* The `M106` command sets a fan speed
* The `P` parameter is used to specify the fan to control by the assigned index of firmware
* The last parameter is fan speed S0 to S255 which equals 0% to 100%

### Syntax

```gcode
M106 PX SXXX ; set fan PX (0-n) to speed SXXX (0-255)
```

### Samples

```gcode
M106 P3 S0 ; set chamber fan to 0% 
M106 P3 S128 ; set chamber fan to 50%
M106 P3 S255 ; set chamber fan to 100%
``` 

![image](https://user-images.githubusercontent.com/67843709/236644845-6f788e99-f775-4e03-8483-94f203df46e4.png)

**Comment by JoshCu** [comment]
I found that I also needed to change the filament specific g-code profile:
![image](https://github-production-user-asset-6210df.s3.amazonaws.com/26720477/238034346-b1ce81e5-4ff5-43e6-ae6b-b96cb5de3906.png)

**Comment by Oliver-S3** [comment]
Just want to add to this. Printing ASA multi color. Tweaked the fan speed to keep the temp where I want it without the smell. Every time it switches filament it resets back to 0 based on the filament gcode. some sort of override would be nice so that you can adjust the chamber fan without having to reslice and cancel the print.

**Comment by michelem09** [comment]
I have a similar issue with BS 1.6.4.2, when I try to set chamber fan, the value come back to the previous one. If you click one time on minus at 100% you got 90 for few seconds then it comes back to 100 and nothing happens fan side. Moreover it seems only few values work on the fan, 30/60/100 or something like that, adjusting it by 10% is impossible.

**Comment by jomiller97** [comment]
How would I set if I wanted to change more than just the chamber fan speed? When I try to make 3 lines just changing the P3 to P1 and P2, it will change the last entry, but then set the other 2 to 0

**Comment by lgarron** [comment]
I'm also finding that this is a big annoyance with my new Bambu printer. It prints excellently and quietly with 70% chamber fan speed, but changing the filament resets the chamber fan to a very noisy 100%.

> Available in v1.5.0 Public Beta

I'm still catching up on this issue, but I'm not quite sure what to make of this. I'm on 1.9.7.52 stable, does this mean this feature should be available? I'm not finding it anywhere in the UI.

`during_print_exhaust_fan_speed` in `fdm_filament_common.json` is set to `70` and the value is not overridden in `fdm_filament_pla.json` and yet I'm still getting 100% fan speed. I can see [some relevant code](https://github.com/bambulab/BambuStudio/blob/4d64b2d65249c141eac0e3cb43712989622012c7/src/libslic3r/GCode.cpp#L1984-L2004) but not any UI for it. Is the only option to modify the G-code as others have suggested above?
